frontends/p4/sideEffects.cpp

```diff
@@ -299,8 +300,9 @@ const IR::Node *DoSimplifyExpressions::preorder(IR::Mux *expression) {
     auto ifFalse = statements;
     statements = save;
 
-    auto ifStatement = new IR::IfStatement(expression->e0, new IR::BlockStatement(ifTrue),
-                                           new IR::BlockStatement(ifFalse));
+    auto ifStatement =
+        new IR::IfStatement(expression->e0->srcInfo, expression->e0, new IR::BlockStatement(ifTrue),
```

Comment:
This one looks questionable to me, but I guess both options work reasonably well.